tokio/tests/signal_info.rs

```diff
@@ -0,0 +1,38 @@
+#![warn(rust_2018_idioms)]
+#![cfg(feature = "full")]
+#![cfg(any(
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "macos",
+    target_os = "netbsd",
+    target_os = "openbsd",
+    target_os = "illumos"
+))]
+#![cfg(not(miri))] // No `sigaction` on Miri
+
+mod support {
+    pub mod signal;
+}
+use support::signal::send_signal;
+
+use tokio::signal;
+use tokio::signal::unix::SignalKind;
+use tokio::sync::oneshot;
+
+#[tokio::test]
+async fn siginfo() {
+    let mut sig = signal::unix::signal(SignalKind::info()).expect("installed signal handler");
+
+    let (fire, wait) = oneshot::channel();
+
+    // NB: simulate a signal coming in by exercising our signal handler
+    // to avoid complications with sending SIGINFO to the test process
+    tokio::spawn(async {
+        wait.await.expect("wait failed");
+        send_signal(libc::SIGINFO);
+    });
+
+    let _ = fire.send(());
+
+    sig.recv().await.expect("received SIGINFO signal");
```

Comment:
Should we consider setting a (very long) timeout on this, so that the test failure mode if the signal is not received is a panic, rather than running forever? Just a thought.

Reply:
I copied the other signal tests which don't set that kind of timeout, but it would certainly make sense to do so. Alternatively, the timeout could be set in nextest. What do you think?

Reply:
We expect tests to run with `cargo test` too, so we shouldn't use nextest specific features. Having a timeout sgtm.

Reply:
All right, done.